### PR TITLE
floating gas price

### PR DESCRIPTION
**Pull request by lsaether**
Changes the restriction that the executing TimeNode must send the same gasPrice as set. Now the requirement is that the set gasPrice is the *minimum* and the TimeNode is allowed to send more. It is not allowed to send less than the set gasPrice.

### CODE REVIEW

**Comment by kosecki123** [comment]
Do we need to test the accounting after this change? Especially the case where scheduler sends more endowment than needed, then how is getting the remaining ETH?

**Comment by lsaether** [comment]
Oh, thats a good point I need to change the accounting to only take into account the set gasPrice and not tx.gasprice. Any gas costs above the set gasPrice will come out of the TimeNode's pocket since it's a voluntary action. There's no reason a scheduler would send higher than the endowment amount, since endowment will always be enough to cover it.

At the end the `sendOwnerEther()` function sends remaining balance to ether

commit incoming...

File: test/transaction-request/accounting.js [hunk 1]
```diff
@@ -387,6 +387,90 @@ contract("Test accounting", async (accounts) => {
...
+    expect(feeAmt).to.equal(fee)
+
+    expect(toBN(afterFeeBal)
+      .sub(toBN(beforeFeeBal))
+      .toNumber()).to.equal(feeAmt)
+
+    const { gasUsed } = executeTx.receipt
+    const gasCost = parseInt(gasUsed, 10) * moreThanRequired
+
+    const expectedBounty = gasCost + requestData.paymentData.bounty
```
Comment (kosecki123) [inline test/transaction-request/accounting.js]: This is a little bit misleading, `expectedBounty` is in fact `gasUsed * gasPrice (set by scheduler) + bounty` which is gas reimbursement + bounty.

**Comment by pipermerriam** [comment]
Just want to point out that in the case of the status token sale, this would create an exploitable vector to force the transaction to fail, since the SNT ICO contract enforced a strict upper limit on the allowed gas prices for purchase.

**Comment by kosecki123** [comment]
Good point @pipermerriam. I did try to figure the best solution for this as a side effect I put together this post https://blog.chronologic.network/how-to-picking-the-right-gas-price-for-scheduled-transactions-94e740328ec9

**Comment by kosecki123** [comment]
Additionally in the case of SNT, assuming claiming and deposit has been requested. The miner should be incentivized to not withhold the tx.

**Pull request merged**